src/entry.c

```diff
@@ -5,63 +5,69 @@
+    struct entry_t* entry;
+    entry = (struct entry_t*) calloc(1,sizeof(struct entry_t));
+    if (entry == NULL || key == NULL || data == NULL)
+        return NULL;
+    entry->key = key;
+    entry->value = data;
+    return entry;
 }
 
-/* Função que elimina uma lista, libertando *toda* a memória utilizada
- * pela lista.
+/* Função que elimina uma entry, libertando a memória por ela ocupada.
  * Retorna 0 (OK) ou -1 em caso de erro.
  */
-int list_destroy(struct list_t *list){
-    
+int entry_destroy(struct entry_t *entry){
+    if(entry == NULL)
+        return -1;
+    free(entry);
+    return 0;
 }
 
-/* Função que adiciona à lista a entry passada como argumento.
- * A entry é inserida de forma ordenada, tendo por base a comparação
- * de entries feita pela função entry_compare do módulo entry e
- * considerando que a entry menor deve ficar na cabeça da lista.
- * Se já existir uma entry igual (com a mesma chave), a entry
- * já existente na lista será substituída pela nova entry,
- * sendo libertada a memória ocupada pela entry antiga.
- * Retorna 0 se a entry ainda não existia, 1 se já existia e foi
- * substituída, ou -1 em caso de erro.
- */
-int list_add(struct list_t *list, struct entry_t *entry);
-
-/* Função que elimina da lista a entry com a chave key, libertando a
- * memória ocupada pela entry.
- * Retorna 0 se encontrou e removeu a entry, 1 se não encontrou a entry,
- * ou -1 em caso de erro.
- */
-int list_remove(struct list_t *list, char *key);
-
-/* Função que obtém da lista a entry com a chave key.
- * Retorna a referência da entry na lista ou NULL se não encontrar a
- * entry ou em caso de erro.
-*/
-struct entry_t *list_get(struct list_t *list, char *key);
-
-/* Função que conta o número de entries na lista passada como argumento.
- * Retorna o tamanho da lista ou -1 em caso de erro.
+/* Função que duplica uma entry, reservando a memória necessária para a
+ * nova estrutura.
+ * Retorna a nova entry ou NULL em caso de erro.
  */
-int list_size(struct list_t *list);
+struct entry_t *entry_dup(struct entry_t *entry){
+    // TODO... Unsure about this
+}
 
-/* Função que constrói um array de char* com a cópia de todas as keys na 
- * lista, colocando o último elemento do array com o valor NULL e
- * reservando toda a memória necessária.
- * Retorna o array de strings ou NULL em caso de erro.
+/* Função que substitui o conteúdo de uma entry, usando a nova chave e
+ * o novo valor passados como argumentos, e eliminando a memória ocupada
+ * pelos conteúdos antigos da mesma.
+ * Retorna 0 (OK) ou -1 em caso de erro.
  */
-char **list_get_keys(struct list_t *list);
+int entry_replace(struct entry_t *entry, char *new_key, struct data_t *new_value){
+    if(entry == NULL || new_key == NULL || new_value == NULL)
+        return -1;
+    free(entry->value);
+    free(entry->key);
+    entry->value = new_value;
+    entry->key = new_key;
+    return 0;
+    //Q: wouldn't it be the same as destroying an entry and recreating it?
```

Comment:
see: comment here
I hope someone can look over it and see if I'm making a correct assumption instead of "not using other existing functions bc I'm a dummy"... I'm not too good with pointers still

Reply:
After thinking about this for a bit, I'm not sure if there would be a difference, practically speaking. Since the `entry` pointer is being passed by value, you could change it _a la_ `entry = new_ptr`, and from the outside the pointer would be updated ig. But I think it's best to leave it like you've coded already.

Reply:
I think I'm going to keep as is :+1: